tests/PerformanceTests/Microsoft.Identity.Web.Perf.Client/TestRunner.cs

```diff
@@ -48,32 +68,76 @@ public async Task Run()
                 }
             }
 
-            // Configuring the http client to trust the self-signed certificate
-            var httpClientHandler = new HttpClientHandler();
-            httpClientHandler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+            IEnumerable<Task> CreateTasks()
+            {
+                // Checks if processing end time has passed
```

Comment:
How does this work? You say that you want to check if processing end time has passed, but if it hasn't you pause for 30s? Sorry, this type of code is hard to read :)

Reply:
Ah, I guess you start a task that simply checks if processing time has been reached. It checks and sleeps for 30s. Ok, makes sense.
Maybe encapsulate this logic in a private method and CreateTask would be more like

yield return StopProcessingByTimeoutTask()
yield return StopProcessingByUserRequestTask()